.github/workflows/build-and-test.yml

```diff
@@ -15,17 +15,20 @@ jobs:
     name: macOS ${{ matrix.configuration }} test build
     runs-on: macos-latest
     strategy:
+      fail-fast: false
       matrix:
         configuration: [debug, release]
     steps:
     - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```

Comment:
@PSchmiedmayer why did you add this line?

Reply:
@tgymnich There was a log error in the job that uploaded the codecov report that the action was requiring a fetch-depth of 0 or bigger then 1 and we had the issue that the report was not always reported in the PRs and after removing the log error the codecov reports continued to work again.

Reply:
thanks for clarifying this!